File: frontend/src/main/frontend/src/app/shared/model/domain/Patient.ts
import {Ingredient} from './Ingredient';
import {Type} from 'class-transformer';

export class Patient
{
  id: number;
  firstName: string;
  lastName: string;
  personalId: string;
  healthCardId: string;

  @Type(() => Ingredient)
  allergicIngredients: Ingredient[];


  constructor();
  constructor(id: number, firstName: string, lastName: string, personalId: string, healthCardId: string, allergicIngredients: Ingredient[]);

  constructor(id?: number, firstName?: string, lastName?: string, personalId?: string, healthCardId?: string, allergicIngredients?: Ingredient[])
  {
    this.id = id;
    this.firstName = firstName;
    this.lastName = lastName;
    this.personalId = personalId;
    this.healthCardId = healthCardId;
    this.allergicIngredients = allergicIngredients;
  }
}

File: backend/src/main/java/rs/ac/uns/ftn/sbz/backend/model/Patient.java
package rs.ac.uns.ftn.sbz.backend.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name = "patient")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Patient extends Person
{
    @Column(name = "patient_health_card_id")
    private String healthCardId;

    @ManyToMany
    @JoinTable(name = "patient_allergic_ingredients",
               joinColumns = { @JoinColumn(name = "person_id") },
               inverseJoinColumns = { @JoinColumn(name = "ingredient_id") })
    private Set<Ingredient> allergicIngredients;

    @OneToMany(mappedBy = "patient")
    private Set<Diagnosis> diagnoses;


    public Patient(Long id, String firstName, String lastName, String personalId, String healthCardId, HashSet<Ingredient> allergicIngredients, HashSet<Diagnosis> diagnoses)
    {
        super(id, firstName, lastName, personalId);

        this.setHealthCardId(healthCardId);
        this.setAllergicIngredients(allergicIngredients);
        this.setDiagnoses(diagnoses);
    }
}
